packages/blockchain/src/consensus/ethash.ts

```diff
@@ -0,0 +1,27 @@
+// eslint-disable-next-line implicit-dependencies/no-implicit
+import type { LevelUp } from 'levelup'
+import { Block } from '@ethereumjs/block'
+import Ethash from '@ethereumjs/ethash'
+import { ConsensusBase } from './base'
+
+/**
+ * This class encapsulates Ethash-related consensus functionality when used with the Blockchain class.
+ */
+export class EthashConsensus extends ConsensusBase {
+  _ethash: Ethash | undefined
+
+  constructor(db: LevelUp) {
+    super(db)
+  }
+
+  async setup(): Promise<void> {
+    this._ethash = new Ethash(this.db)
+  }
+
+  async validate(block: Block): Promise<void> {
+    const valid = await this._ethash!.verifyPOW(block)
```

Comment:
Would it make sense to check if `ethash` is defined and explicitly return an error that you need to trigger setup method first?

Reply:
yeah, well even better ethash should not be optionally undefined in this class, ideally it would be strictly defined, can you do the ethash assignment currently in `setup()` in the constructor in this case? then it should let you removed the `| undefined`

Reply:
Just read through Holger's comments on the issue more closely, and looks like there's no issue with moving the initialization to the constructor. Will update 👍